beacon-chain/core/state/transition.go

```diff
@@ -272,10 +274,18 @@ func ProcessSlots(ctx context.Context, state iface.BeaconState, slot types.Slot)
 			return nil, errors.Wrap(err, "could not process slot")
 		}
 		if CanProcessEpoch(state) {
-			state, err = ProcessEpochPrecompute(ctx, state)
-			if err != nil {
-				traceutil.AnnotateError(span, err)
-				return nil, errors.Wrap(err, "could not process epoch with optimizations")
+			if state.Version() == version.Altair {
+				state, err = altair.ProcessEpoch(ctx, state)
+				if err != nil {
+					traceutil.AnnotateError(span, err)
+					return nil, errors.Wrap(err, "could not process epoch with optimizations")
```

Comment:
I have a feeling that `with optimizations` should only apply to the `else` branch because we don't call `ProcessEpochPrecompute` here.